File: JS/LeetC/arrays/validMountainArray.js
function validMountainArray(arr) {
	let i = 0;

	while (i < arr.length) {

		if (arr[i] >= arr[i + 1]) {
			break;
		}

		i++;
	}

	if (i === 0 || i >= arr.length) {
		return false;
	}

	while (i < arr.length) {
		if (arr[i] <= arr[i + 1]) {
			return false;
		}

		i++;
	}

	return true;
}

console.log(validMountainArray([2, 1]));

File: python/leetcode/arrays/validMountainArray.py
class Solution:

    def validMountainArray(self, arr):

        i = 0
        l = len(arr)

        while i < l - 1 and arr[i] < arr[i + 1]:
            i += 1

        if i == 0 or i >= l - 1:
            return False

        while i < l - 1 and arr[i] > arr[i + 1]:
            i += 1

        return i == l - 1


s = Solution()

arr = [2, 1]

print(s.validMountainArray(arr))
